addons/website_sale/models/sale_order.py

```diff
@@ -426,13 +358,100 @@ def _cart_find_product_line(
 
         return filtered_sol
 
+    def _cart_update_line_quantity(self, line_id:int, quantity:int|float, **kwargs):
+        """Update the quantity of a given line of the cart.
+
+        :param int line_id: line id, as a `sale.order.line` id.
+        :params float quantity: the updated quantity of the line.
+        :param dict kwargs: Additional parameters given to deeper method calls.
+        :return: values used by the cart service to give feedback to the customer.
+        :rtype: dict
+        """
+        self.ensure_one()
+        self = self.with_company(self.company_id)
+
+        if not (order_line := self.order_line.filtered(lambda sol: sol.id == line_id)):
+            raise UserError(_("This line doesn't belong to your order."))
+
+        if quantity > 0:
+            quantity, warning = self._verify_updated_quantity(
+                order_line,
+                order_line.product_id.id,
+                quantity,
+                **kwargs,
+            )
+        else:
+            # If the line will be removed anyway, there is no need to verify
+            # the requested quantity update.
+            warning = ''
+
+        order_line = self._cart_update_order_line(order_line, quantity, **kwargs)
```

Comment:
I'd prefer keeping that in `_cart_add`, it's not that complex and I'd like to avoid having two places where we call `_verify_cart_after_update` and return data (I fear we would forgot to update this method if we change the return of `_cart_add`, it kinda defeats the one entry point)